### PR TITLE
Change custom request types to LSP commands

### PR DESCRIPTION
**Pull request by remcohaszing**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

LSP commands are the intended way to implement custom behaviour.

### CODE REVIEW

**Review by ChristianMurphy — COMMENTED** [review]
The tests look like they're failing 🤔

File: packages/language-service/lib/commands.js [hunk 1]
```diff
@@ -146,3 +139,12 @@ export function createSyntaxToggle(context, type, separator) {
     }
   }
 }
+
+export const implementations = {
+  'mdx.toggleDelete': createSyntaxToggle('delete', '~'),
+  'mdx.toggleEmphasis': createSyntaxToggle('emphasis', '_'),
```
Comment (wooorm) [inline packages/language-service/lib/commands.js:145]: `*`

File: packages/language-service/lib/commands.js [hunk 2]
```diff
@@ -146,3 +139,12 @@ export function createSyntaxToggle(context, type, separator) {
     }
   }
 }
+
+export const implementations = {
+  'mdx.toggleDelete': createSyntaxToggle('delete', '~'),
```
Comment (wooorm) [inline packages/language-service/lib/commands.js:144]: where’s this word from, `delete`? Perhaps `strikethrough` is better. Also, it’s a GFM-only feature. And, 2 tildes work there too. One tilde was forbidden by the GFM spec for a while, until recently, even though it worked.

Comment (remcohaszing) [reply packages/language-service/lib/commands.js:144]: `delete` is the mdast Node type. Based on your comment, I agree inserting 2 tildes is better. This changed be done in a follow-up PR though, as it’s an unrelated change.

File: packages/language-service/lib/commands.js (hunk 1 above)
Comment (remcohaszing) [reply packages/language-service/lib/commands.js:145]: It doesn’t really matter. I know remark stringifies uses `*` in its output by default. I personally feel `*` is a bit overloaded. Emphasis and strong text share the same characters, but `*` is also used for list items. Also Prettier uses `_`, which I think added to making that character option more popular.

TL;DR I prefer `_`, but we can change it to `*` if you feel very strongly about it. This should be done in a follow-up PR though, as it’s an unrelated change.

**Comment by remcohaszing** [comment]
> The tests look like they're failing 🤔

Yes, because of a timing issue. I think we were just lucky the tests weren’t failing before. I’ll look into it.

File: packages/language-service/lib/commands.js (hunk 1 above)
Comment (wooorm) [reply packages/language-service/lib/commands.js:145]: Why does it not matter? Yes, it matters. `_` and `*` get parsed differently.

File: packages/language-service/lib/commands.js (hunk 2 above)
Comment (wooorm) [reply packages/language-service/lib/commands.js:144]: This doesn’t particularly have to match the node type, as it’s injecting into a document.

File: packages/language-service/lib/commands.js (hunk 1 above)
Comment (remcohaszing) [reply packages/language-service/lib/commands.js]: Both exist, but have relatively rare use cases in natural language. It’s really just a preference, and it’s ok your preference differs from mine.

I will change it in a follow-up PR anyway, as I believe the strongest argument is that it’s good to match the `remark-stringify` default. If we enable configurable formatting for #456, I think we should respect the same configuration optionn for syntax toggles.

File: packages/language-service/lib/commands.js (hunk 2 above)
Comment (remcohaszing) [reply packages/language-service/lib/commands.js]: It’s a toggle. It either injects or removes, depending on whether the cursor position matches a node of this type.

**Comment by remcohaszing** [comment]
I’ll merge this as-is and process the comments in follow-up PRs.
